### PR TITLE
[PoC] Improve References extension

### PR DESCRIPTION
**Pull request by aramalipoor**
I needed **XML mapping** driver for `References` and also **Phpcr support**, so I've added a few lines of code.

I'm very no0b in Doctrine and Symfony so I'm not sure if all cases are covered here!

### CODE REVIEW

**Comment by aramalipoor** [comment]
@l3pp4rd Can you please kindly help with the failed test? I don't exactly know what is going on here! :)

**Comment by l3pp4rd** [comment]
well, I have no idea, haven't used this extension ever :) try looking if the test is correct in the first place and why the assertion fails. I really do not want to waste time here, it takes long to debug ORM and looking for behavioral reasons..

File: lib/Gedmo/References/Mapping/Driver/Xml.php [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    {
+        /**
+         * @var \SimpleXmlElement $xml
+         */
+        $xml = $this->_getMapping($meta->name);
+        $xmlDoctrine = $xml;
+
+        $xml = $xml->children(self::GEDMO_NAMESPACE_URI);
+
+        if ($xmlDoctrine->getName() == 'entity' || $xmlDoctrine->getName() == 'document' || $xmlDoctrine->getName() == 'mapped-superclass') {
```
Comment (stof) [inline lib/Gedmo/References/Mapping/Driver/Xml.php]: please use strict comparisons

**Comment by aramalipoor** [comment]
@stof Thanks for the review.
@l3pp4rd Is it OK to merge?

**Comment by aramalipoor** [comment]
@stof Thanks for your input :)
@l3pp4rd I think it's ready to merge ;)

**Comment by l3pp4rd** [comment]
thanks ;)

**Pull request merged**